packages/strapi-plugin-graphql/hooks/graphql/create-subscription-server.js

```diff
@@ -0,0 +1,50 @@
+const _ = require('lodash');
+const { SubscriptionServer } = require('subscriptions-transport-ws');
+const { execute, subscribe } = require('graphql');
+
+const getUserFromParams = async connectionParams => {
+  if (_.isEmpty(connectionParams) || _.isEmpty(connectionParams.Authorization)) {
+    throw new Error('Missing Bearer Token');
+  }
+
+  const token = await strapi.plugins['users-permissions'].services.jwt.retrieveToken(connectionParams.Authorization);
+  const { id, _id } = await strapi.plugins['users-permissions'].services.jwt.verify(token);
+
+  if ((id || _id) === undefined) {
+    throw new Error('Invalid token: Token did not contain required fields');
+  }
+
+  const user =  strapi.plugins['users-permissions'].models.user.findOne({ _id, id });
```

Comment:
You should use our Query system because the `findOne` method only works with MongoDB. It won't work with SQL databases (Bookshelf).

```js
const user = strapi.query('user', 'users-permissions').findOne({ _id, id });
```

Then, you'll have to write the `findOne` query for both ORM (bookshelf and mongoose).

Reply:
Good call